### PR TITLE
Housekeeping -  Upgrade dependencies

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- https://github.com/RasaHQ/rasa/pull/12727
- https://github.com/RasaHQ/rasa/pull/12698
- https://github.com/RasaHQ/rasa/pull/12594
- https://github.com/RasaHQ/rasa/pull/12509
- https://github.com/RasaHQ/rasa/pull/12369

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by vcidst** [comment]
Is this a duplicate of https://github.com/RasaHQ/rasa/pull/12740 it seems like engine-atom both do the same upgrades sometimes 🤔

**Comment by Tawakalt** [comment]
No it's not.
[12740](https://github.com/RasaHQ/rasa/pull/12740) only upgrades a single dependency. We can close that once this gets merged.

File: pyproject.toml [hunk 1]
```diff
@@ -131,7 +131,7 @@ sklearn-crfsuite = "~0.3"
 psycopg2-binary = ">=2.8.2,<2.10.0"
 python-dateutil = "~2.8"
 protobuf = ">=4.23.3,< 4.23.4"
-tensorflow_hub = "^0.13.0"
+tensorflow_hub = "^0.14.0"
```
Comment (Urkem) [inline pyproject.toml:134]: Isn't engine supposed to do ML package upgrades?

Comment (Tawakalt) [reply pyproject.toml:134]: You're right!

**Review by Urkem — APPROVED** [review]

**Pull request merged**